admin_manual/configuration/server/security/oauth2.rst

```diff
@@ -0,0 +1,174 @@
+The client redirects the resource owner to the authorization URL and requests authorization.
+The following URL parameters have to be specified:
+
+================= ========== ========================================================================================
+Parameter         Required   Description
+================= ========== ========================================================================================
+``response_type``  yes       Needs to be `code` because at this time only the authorization code flow is implemented.
+``client_id``      yes       The client identifier obtained when registering the client.
+``redirect_uri``   yes       The redirection URI specified when registering the client.
+``state``          no        Can be set by the client "to maintain state between the request and callback".
+                             See `RFC 6749`_ for more information.
+================= ========== ========================================================================================
+
+For further information about client registration, please refer to `the official authorization request RFC from the IETF`_.
+
+Authorization Response
+^^^^^^^^^^^^^^^^^^^^^^^
+
+After the resource owner's authorization, the app redirects to the `redirect_uri` specified in the authorization request and adds the authorization code as URL parameter `code`.
+An authorization code is valid for 10 minutes.
+For further information about client registration, please refer to `the official authorization response RFC from the IETF`_.
+
+Access Token Request
+^^^^^^^^^^^^^^^^^^^^
+
+With the authorization code, the client can request an access token using the access token URL.
+`Client authentication`_ is done using basic authentication with the client identifier as username and the client secret as a password.
+The following URL parameters have to be specified:
+
+================= ============================= ===================================================
+Parameter         Required                      Description
+================= ============================= ===================================================
+``grant_type``                                  Either ``authorization_code`` or ``refresh_token``.
+``code``          if the grant type
+                  `authorization_code` is used.
+``redirect_uri``  if the grant type
+                  `authorization_code` is used.
+``refresh_token`` if the grant type
+                  `refresh_token` is used.
+================= ============================= ===================================================
+
+For further information about client registration, please refer to `the official access token request RFC from the IETF`_.
+
+Access Token Response
+^^^^^^^^^^^^^^^^^^^^^
+
+The app responses to a valid access token request with a JSON response like the following.
+An access token is valid for 1 hour and can be refreshed with a refresh token.
```

Comment:
Above, the token is valid for 10min. Now the token is valid for 1h...
Additionally same question as above, what happens if it expires... Can be refreshed means what ?

Reply:
I guess above is referring to the authorization code, with this being valid for 10 minutes. An authorization code can then be used for requesting an access token, with this being valid for max 1 hour. Or at least this is what I understood..

Reply:
I'd suggest there's some confusion between authorization code and access token going on. [Here's a good description of them](https://stackoverflow.com/a/16341985/222011).

Reply:
Why not adding the link (good description) to ease understanding?

Reply:
Done!